SU2_CFD/src/solvers/CNEMONSSolver.cpp

```diff
@@ -57,17 +57,20 @@ CNEMONSSolver::CNEMONSSolver(CGeometry *geometry, CConfig *config, unsigned shor
 void CNEMONSSolver::Preprocessing(CGeometry *geometry, CSolver **solver_container, CConfig *config, unsigned short iMesh,
                               unsigned short iRKStep, unsigned short RunTime_EqSystem, bool Output) {
 
-  const unsigned long InnerIter = config->GetInnerIter();
-  const bool limiter    = (config->GetKind_SlopeLimit_Flow() != NO_LIMITER) && (InnerIter <= config->GetLimiterIter());
+  const auto InnerIter = config->GetInnerIter();
+  const bool limiter = (config->GetKind_SlopeLimit_Flow() != NO_LIMITER) && (InnerIter <= config->GetLimiterIter());
   const bool van_albada = config->GetKind_SlopeLimit_Flow() == VAN_ALBADA_EDGE;
-  const bool muscl      = config->GetMUSCL_Flow() && (iMesh == MESH_0);
-  const bool center     = config->GetKind_ConvNumScheme_Flow() == SPACE_CENTERED;
+  const bool muscl = config->GetMUSCL_Flow() && (iMesh == MESH_0);
+  const bool center = config->GetKind_ConvNumScheme_Flow() == SPACE_CENTERED;
+  const bool wall_functions = config->GetWall_Functions();
 
   /*--- Common preprocessing steps (implemented by CNEMOEulerSolver) ---*/
 
   CommonPreprocessing(geometry, solver_container, config, iMesh, iRKStep, RunTime_EqSystem, Output);
 
   /*--- Compute gradient for MUSCL reconstruction. ---*/
+  const auto nPrimVarGrad_bak = nPrimVarGrad;
+  if (Output) ompMasterAssignBarrier(nPrimVarGrad, 1+nDim);
```

Comment:
I don't think this optimization is valid for NEMO, in the EulerSolver we can consider only the temperature and velocities, but NEMO has a lot more stuff before the velocities.

Reply:
This one.

Reply:
I removed this.